wpilib/wpilib/interruptablesensorbase.py

```diff
@@ -97,28 +90,33 @@ def waitForInterrupt(self, timeout, ignorePrevious=True):
         :param ignorePrevious: If True (default), ignore interrupts that
             happened before waitForInterrupt was called.
         """
-        if self.interrupt is None:
+        if self.interrupt is 0:
             raise ValueError("The interrupt is not allocated.")
         result = hal.waitForInterrupt(self.interrupt, timeout, ignorePrevious)
         rising = 0x1 if (result & 0xFF) else 0x0
         falling = 0x100 if (result & 0xFF00) else 0x0
 
-        return rising | falling
+        result = rising | falling
+        for mode in self.WaitResult:
+            if mode == result:
+                return mode
+
+        return None
```

Comment:
Seeing as you've made WaitResult an enum, use `self.WaitResult(result)` instead.